collect_app/src/main/java/org/odk/collect/android/widgets/utilities/DateTimeWidgetUtils.java

```diff
@@ -0,0 +1,124 @@
+import org.odk.collect.android.fragments.dialogs.BikramSambatDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.CopticDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.CustomDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.CustomTimePickerDialog;
+import org.odk.collect.android.fragments.dialogs.EthiopianDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.FixedDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.IslamicDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.MyanmarDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.PersianDatePickerDialog;
+import org.odk.collect.android.logic.DatePickerDetails;
+import org.odk.collect.android.utilities.DateTimeUtils;
+import org.odk.collect.android.utilities.DialogUtils;
+import org.odk.collect.android.utilities.ThemeUtils;
+
+import java.util.Date;
+
+import static org.odk.collect.android.fragments.dialogs.CustomDatePickerDialog.DATE_PICKER_DIALOG;
+import static org.odk.collect.android.fragments.dialogs.CustomTimePickerDialog.CURRENT_TIME;
+import static org.odk.collect.android.fragments.dialogs.CustomTimePickerDialog.TIME_PICKER_THEME;
+import static org.odk.collect.android.fragments.dialogs.FixedDatePickerDialog.CURRENT_DATE;
+import static org.odk.collect.android.fragments.dialogs.FixedDatePickerDialog.DATE_PICKER_DETAILS;
+import static org.odk.collect.android.fragments.dialogs.FixedDatePickerDialog.DATE_PICKER_THEME;
+
+public class DateTimeWidgetUtils {
+
+    private DateTimeWidgetUtils() {
+    }
+
+    public static TimeData getTimeData(int hourOfDay, int minuteOfHour, boolean nullAnswer) {
+        // use picker time, convert to today's date, store as utc
+        DateTime localDateTime = new DateTime()
+                .withTime(hourOfDay, minuteOfHour, 0, 0);
+
+        return !nullAnswer
+                ? new TimeData(localDateTime.toDate())
+                : null;
+    }
+
+    public static void setDateLabel(Context context, TextView dateAnswerText, Date date, DatePickerDetails datePickerDetails) {
+        dateAnswerText.setText(DateTimeUtils.getDateTimeLabel(date, datePickerDetails, false, context));
+    }
+
+    public static void setTimeLabel(TextView timeAnswerText, int hourOfDay, int minuteOfHour, boolean nullAnswer) {
+        timeAnswerText.setText(getTimeData(hourOfDay, minuteOfHour, nullAnswer).getDisplayText());
+    }
+
+    public static void showDatePickerDialog(FormEntryActivity activity, FormEntryPrompt prompt, DatePickerDetails datePickerDetails,
+                                            LocalDateTime date) {
+        switch (datePickerDetails.getDatePickerType()) {
+            case ETHIOPIAN:
+                CustomDatePickerDialog dialog = EthiopianDatePickerDialog.newInstance(prompt.getIndex(), date, datePickerDetails);
+                dialog.show(activity.getSupportFragmentManager(), DATE_PICKER_DIALOG);
```

Comment:
i think we can get rid of this `DATE_PICKER_DIALOG` static value and use `CustomDatePickerDialog.class.getName()` in all cases.

Reply:
And also i'm wondering if we can simplify it somehow like:
```
    public static void showDatePickerDialog(FormEntryActivity activity, FormEntryPrompt prompt, DatePickerDetails datePickerDetails,
                                            LocalDateTime date) {

        Bundle bundle = new Bundle();
        bundle.putInt(DATE_PICKER_THEME, getTheme(themeUtils, datePickerDetails));
        bundle.putSerializable(CURRENT_DATE, date);
        bundle.putSerializable(DATE_PICKER_DETAILS, datePickerDetails);

        DialogUtils.showIfNotShowing(getClass(datePickerDetails.getDatePickerType()), bundle, activity.getSupportFragmentManager());
    }
    
    private static Class getClass(DatePickerDetails.DatePickerType datePickerType) {
        switch (datePickerType) {
            case ETHIOPIAN:
                return CustomDatePickerDialog.class;
            case COPTIC:
                return CopticDatePickerDialog.class;
            case ISLAMIC:
                return IslamicDatePickerDialog.class;
            case BIKRAM_SAMBAT:
                return BikramSambatDatePickerDialog.class;
            case MYANMAR:
                return MyanmarDatePickerDialog.class;
            case PERSIAN:
                return PersianDatePickerDialog.class;
            default:
                return FixedDatePickerDialog.class;
        }
    }
```

you would need to add some fixes in handling those values we pass to dialogs but I think it's possible?

Reply:
Ohh that's great! I'll work on it.